admin/includes/modules/update_product.php

```diff
@@ -15,40 +15,33 @@
-                            'products_weight' => $products_weight,
-                            'products_status' => zen_db_prepare_input((int)$_POST['products_status']),
-                            'products_virtual' => zen_db_prepare_input((int)$_POST['products_virtual']),
-                            'products_tax_class_id' => zen_db_prepare_input((int)$_POST['products_tax_class_id']),
-                            'manufacturers_id' => $manufacturers_id,
-                            'products_quantity_order_min' => zen_db_prepare_input(($_POST['products_quantity_order_min'] == 0 ? 1 : $_POST['products_quantity_order_min'])),
-                            'products_quantity_order_units' => zen_db_prepare_input(($_POST['products_quantity_order_units'] == 0 ? 1 : $_POST['products_quantity_order_units']) ),
-                            'products_priced_by_attribute' => zen_db_prepare_input((int)$_POST['products_priced_by_attribute']),
-                            'product_is_free' => zen_db_prepare_input((int)$_POST['product_is_free']),
-                            'product_is_call' => zen_db_prepare_input((int)$_POST['product_is_call']),
-                            'products_quantity_mixed' => zen_db_prepare_input($_POST['products_quantity_mixed']),
-                            'product_is_always_free_shipping' => zen_db_prepare_input((int)$_POST['product_is_always_free_shipping']),
-                            'products_qty_box_status' => zen_db_prepare_input($_POST['products_qty_box_status']),
-                            'products_quantity_order_max' => zen_db_prepare_input($_POST['products_quantity_order_max']),
-                            'products_sort_order' => (int)zen_db_prepare_input($_POST['products_sort_order']),
-                            'products_discount_type' => zen_db_prepare_input($_POST['products_discount_type']),
-                            'products_discount_type_from' => zen_db_prepare_input($_POST['products_discount_type_from']),
-                            'products_price_sorter' => zen_db_prepare_input($_POST['products_price_sorter'])
-                            );
+
+    // Data-cleaning to prevent data-type mismatch errors:
+        $sql_data_array = array(
+            'products_quantity' => convertToFloat($_POST['products_quantity']),
+            'products_type' => (int)$_GET['product_type'],
+            'products_model' => zen_db_prepare_input($_POST['products_model']),
+            'products_price' => convertToFloat($_POST['products_price']),
+            'products_date_available' => $products_date_available,
+            'products_weight' => convertToFloat($_POST['products_weight']),
+            'products_status' => (int)$_POST['products_status'],
+            'products_virtual' => (int)$_POST['products_virtual'],
+            'products_tax_class_id' => (int)$_POST['products_tax_class_id'],
+            'manufacturers_id' => convertToFloat($_POST['manufacturers_id']),

```

Comment:
Shouldn't this (manufacturers_id) be an (int) cast instead (unless there is a move afoot to have something like a Dewey Decimal Classification system)? (not that it is likely to cause a problem (famous last words) by being a float.)

Reply:
Yup. Good catch. Got lost in the sea of fields...

Reply:
Updated.